config/types/locksmith.go

```diff
@@ -33,51 +33,47 @@ var (
 )
 
 type Locksmith struct {
-	RebootStrategy RebootStrategy `yaml:"reboot_strategy" locksmith:"REBOOT_STRATEGY"`
-	WindowStart    WindowStart    `yaml:"window_start"    locksmith:"LOCKSMITHD_REBOOT_WINDOW_START"`
-	WindowLength   WindowLength   `yaml:"window_length"   locksmith:"LOCKSMITHD_REBOOT_WINDOW_LENGTH"`
-	Group          string         `yaml:"group"           locksmith:"LOCKSMITHD_GROUP"`
-	EtcdEndpoints  string         `yaml:"etcd_endpoints"  locksmith:"LOCKSMITHD_ENDPOINT"`
-	EtcdCAFile     string         `yaml:"etcd_cafile"     locksmith:"LOCKSMITHD_ETCD_CAFILE"`
-	EtcdCertFile   string         `yaml:"etcd_certfile"   locksmith:"LOCKSMITHD_ETCD_CERTFILE"`
-	EtcdKeyFile    string         `yaml:"etcd_keyfile"    locksmith:"LOCKSMITHD_ETCD_KEYFILE"`
+	RebootStrategy string `yaml:"reboot_strategy" locksmith:"REBOOT_STRATEGY"`
+	WindowStart    string `yaml:"window_start"    locksmith:"LOCKSMITHD_REBOOT_WINDOW_START"`
+	WindowLength   string `yaml:"window_length"   locksmith:"LOCKSMITHD_REBOOT_WINDOW_LENGTH"`
+	Group          string `yaml:"group"           locksmith:"LOCKSMITHD_GROUP"`
+	EtcdEndpoints  string `yaml:"etcd_endpoints"  locksmith:"LOCKSMITHD_ENDPOINT"`
+	EtcdCAFile     string `yaml:"etcd_cafile"     locksmith:"LOCKSMITHD_ETCD_CAFILE"`
+	EtcdCertFile   string `yaml:"etcd_certfile"   locksmith:"LOCKSMITHD_ETCD_CERTFILE"`
+	EtcdKeyFile    string `yaml:"etcd_keyfile"    locksmith:"LOCKSMITHD_ETCD_KEYFILE"`
 }
 
 func (l Locksmith) configLines() []string {
-	return getArgs("%s=%q", "locksmith", l)
+	return getArgs("%s=%v", "locksmith", l)
 }
 
-type RebootStrategy string
-type WindowStart string
-type WindowLength string
-
 func (l Locksmith) Validate() report.Report {
 	if (l.WindowStart != "" && l.WindowLength == "") || (l.WindowStart == "" && l.WindowLength != "") {
 		return report.ReportFromError(ErrMissingStartOrLength, report.EntryError)
 	}
 	return report.Report{}
 }
 
-func (r RebootStrategy) Validate() report.Report {
-	switch strings.ToLower(string(r)) {
+func (l Locksmith) ValidateRebootStrategy() report.Report {
```

Comment:
What ensures that this function is called now?

Reply:
The validate code checks for functions on structs with `Validate<NAME>` where `<NAME>` is the name of the field, and calls them if they exist.

Here's the logic that does that: https://github.com/coreos/ignition/blob/58bcc85a20872686c2d6d2f8db2e439cd2e1ccb5/config/validate/validate.go#L185

Reply:
Neat.